### PR TITLE
Add a contributing guide

### PR DESCRIPTION
**Pull request by tomassedovic**
#### What does this PR do?
It adds a contributing guide.

Hoping that this will get our pull requests and code reviews more
consistent (and it should help people getting started with code
reviews).

#### How should this be manually tested?
It's just a docs change, no testing required

#### Is there a relevant Issue open for this?
n/a

#### Who would you like to review this?
cc: @Tlacenka @tzumainn  @bogdando PTAL

### CODE REVIEW

**Comment by tzumainn** [comment]
:+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Contributing
+
+## Submitting Code To The Playbooks
+
+Following these guidelines will it easier for people to review your
```
Comment (bogdando) [inline CONTRIBUTING.md]: will make it easier

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+review process faster.
+
+When you do a rebase, we can't see what changed before the last
+revision and have to look at the entire pull request again -- even if
+all you added was a small documentation fix!
+
+If you need to bring in changes from the master, you can do `git merge
+master` instead or rebasing.
+
+When we merge the pull request, we will squash your commits so don't
```
Comment (bogdando) [inline CONTRIBUTING.md]: Let's please leave squashing to the contributors, once asked to do so.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+Make the change as an new commit instead and push it to your branch.
+That way, Github lets us see just the new changes, which will make the
+review process faster.
+
+When you do a rebase, we can't see what changed before the last
+revision and have to look at the entire pull request again -- even if
+all you added was a small documentation fix!
+
+If you need to bring in changes from the master, you can do `git merge
+master` instead or rebasing.
```
Comment (bogdando) [inline CONTRIBUTING.md]: Rebasing changes and resolving conflicts produces a better looking history than master merges, but I can live with that.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tomassedovic) [inline CONTRIBUTING.md]: I mean the "Squash and merge" button in github. When you click merge it squashes the changes and leaves a nice history.

See the bottom of this article: https://github.com/blog/2141-squash-your-commits

File: CONTRIBUTING.md (hunk 1 above)
Comment (tomassedovic) [inline CONTRIBUTING.md]: Thanks! Done.

**Pull request merged**